# HG changeset patch
# User Rik <rik@octave.org>
# Date 1344626399 25200
#      Fri Aug 10 12:19:59 2012 -0700
# Node ID f4684192158ec391b9bc9994e813bbebf7a2882d
# Parent  4388f651844097b01b0349ea4e15a3bf88962718
configure.ac: Split gnulib initialization into 2 parts to match recommended usage.

* configure.ac: Split gnulib initialization into 2 parts to match
recommended usage.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -261,20 +261,20 @@ case "$canonical_host_type" in
 #ifndef _MSC_VER
 #error "Not MSVC compiler"
 #endif
 ])], have_msvc=yes, have_msvc=no)
     AC_MSG_RESULT([$have_msvc])
   ;;
 esac
 
-### gnulib initialization
+### gnulib initialization: part 1
+### Must take place immediately after a compiler is determined
 
 gl_EARLY
-gl_INIT
 
 ### Check version number when using gcc.
 
 gcc_version=`$CC -v 2>&1 | grep "^.*gcc version" | \
   sed -e 's/^.*g.. version *//' -e 's/cygnus-//' -e 's/egcs-//'`
 case "$gcc_version" in
   [12].*)
     warn_gcc_version="gcc version $gcc_version is likely to cause problems"
@@ -1860,16 +1860,22 @@ OCTAVE_CXX_COMPLEX_REFERENCE_ACCESSORS
 OCTAVE_FAST_INT_OPS
 
 ## Does the C compiler handle alloca() and const correctly?
 AC_FUNC_ALLOCA
 
 ## Does the C compiler support Automake subdir-objects option?
 AM_PROG_CC_C_O 
 
+### gnulib initialization: part 2
+### After all include and path modifications have taken place
+### and at the same priority level as function checks.
+
+gl_INIT
+
 ### Checks for functions and variables.
 
 AC_CHECK_FUNCS(basename canonicalize_file_name \
   chmod dup2 endgrent endpwent execvp expm1 expm1f fork \
   getegid geteuid getgid getgrent getgrgid getgrnam getpgrp getpid \
   getppid getpwent getpwuid getuid getwd _kbhit kill \
   lgamma lgammaf lgamma_r lgammaf_r localtime_r log1p log1pf \
   mkstemp pipe putenv \
